core/src/main/java/org/elasticsearch/index/get/GetField.java

```diff
@@ -116,9 +116,14 @@ public static GetField fromXContent(XContentParser parser) throws IOException {
         String fieldName = parser.currentName();
         XContentParser.Token token = parser.nextToken();
         ensureExpectedToken(XContentParser.Token.START_ARRAY, token, parser::getTokenLocation);
+
         List<Object> values = new ArrayList<>();
         while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-            values.add(parseStoredFieldsValue(parser));
+            if (token.isValue()) {
+                values.add(parseStoredFieldsValue(parser));
+            } else {
+                parser.skipChildren();
```

Comment:
I also thought looking below that we are not testing this at the moment, maybe I am wrong.